components/brave_wallet_ui/components/desktop/portfolio_transaction_item/portfolio_transaction_item.tsx

```diff
@@ -0,0 +1,690 @@
+  useGetCombinedTokensListQuery
+} from '../../../common/slices/api.slice.extra'
+
+// Components
+import {
+  NftIcon
+} from '../../shared/nft-icon/nft-icon'
+import {
+  Skeleton
+} from '../../shared/loading-skeleton/styles'
+import {
+  withPlaceholderIcon
+} from '../../shared/create-placeholder-icon'
+import {
+  TransactionDetailsModal
+} from '../popup-modals/transaction_details_modal/transaction_details_modal'
+import {
+  CreateNetworkIcon
+} from '../../shared/create-network-icon'
+
+// Styled Components
+import {
+  PortfolioTransactionItemWrapper,
+  TransactionTypeIcon,
+  DateText,
+  TransactionTypeText,
+  IntentAddressText,
+  AssetIcon,
+  SwapIcon,
+  SwapSellIcon,
+  SwapBuyIcon,
+  TokenNameText,
+  TokenSymbolText,
+  AssetValueText,
+  FiatValueText,
+  ArrowIcon,
+  ArrowIconWrapper,
+  StatusBubble,
+  StatusIcon,
+  LoadingIcon,
+  BalancesColumn,
+  NetworkIconWrapper,
+  IconWrapper,
+  SwapIconsWrapper,
+  SellIconPlaceholder,
+  BuyIconPlaceholder,
+  SwapPlaceholderIcon
+} from './portfolio_tranasction_item.style'
+import {
+  Column,
+  Row,
+  VerticalSpace
+} from '../../shared/style'
+
+export interface Props {
+  transaction: BraveWallet.TransactionInfo | SerializableTransactionInfo
+  isFocused?: boolean
+}
+
+const ICON_ASSET_CONFIG =
+  { size: 'medium', marginLeft: 0, marginRight: 0 } as const
+const ICON_SWAP_CONFIG =
+  { size: 'small', marginLeft: 0, marginRight: 0 } as const
+const ICON_NFT_CONFIG =
+  { size: 'medium', marginLeft: 0, marginRight: 0 } as const
+const AssetIconWithPlaceholder =
+  withPlaceholderIcon(AssetIcon, ICON_ASSET_CONFIG)
+const SwapIconWithPlaceholder =
+  withPlaceholderIcon(SwapIcon, ICON_SWAP_CONFIG)
+const NftIconWithPlaceholder =
+  withPlaceholderIcon(NftIcon, ICON_NFT_CONFIG)
+
+export const PortfolioTransactionItem =
+  React.forwardRef<HTMLDivElement, Props>(({
+    transaction,
+    isFocused
+  }: Props, forwardedRef) => {
+
+    // State
+    const [showTransactionDetails, setShowTransactionDetails]
+      = React.useState<Boolean>(false)
+
+    // partial tx parsing
+    const {
+      isSolanaTx,
+      recipient,
+      approvalTarget,
+      isSwap,
+      txCoinType,
+    } = React.useMemo(() => {
+      return {
+        isSolanaTx: isSolanaTransaction(transaction),
+        recipient: getTransactionToAddress(transaction),
+        approvalTarget: getTransactionApprovalTargetAddress(transaction),
+        isSwap: isSwapTransaction(transaction),
+        txCoinType: getCoinFromTxDataUnion(transaction.txDataUnion),
+      }
+    }, [transaction])
+
+    const {
+      data: defaultFiatCurrency = '',
+      isLoading: isLoadingDefaultFiatCurrency
+    } = useGetDefaultFiatCurrencyQuery(undefined)
+
+    const {
+      data: txNetwork,
+    } = useGetNetworkQuery({
+      chainId: transaction.chainId,
+      coin: txCoinType
+    })
+
+    const networkAsset = React.useMemo(() => {
+      return makeNetworkAsset(txNetwork)
+    }, [txNetwork])
```

Comment:
Nit: move to "computed from queries" section

Reply:
Moved